src/shapes/Group.ts

```diff
@@ -431,7 +357,7 @@ export class Group extends createCollectionMixin(
     this._shouldSetNestedCoords() && object.setCoords();
     object._set('group', this);
     object._set('canvas', this.canvas);
-    this.interactive && this._watchObject(true, object);
+    this._watchObject(true, object);
```

Comment:
why watching if the group is not interactive?

Reply:
1. I think we can remove the watching functionality because it is not used and can be replaced by `canvas.getActiveObjects().includes(obj)`
2. the constructor was throwing because it was setting this option before populationg `_objects` - but that can be fixed if we do not pass options to super and set them afterwards.
3. the selected/deselected events will fire only if the group is interactive so this is a redundant check making the use of interactive more annoying because if you toggle it without using `set` the group will be in a broken state.